### PR TITLE
Cherry pick fixes

### PR DESCRIPTION
**Pull request by SvenDowideit**
prepare for a docs update - can you please confirm that these are all fixes to the existing release's docs, and that I've not missed one?

There are 3 PR's that I'm not sure about, but didn't look like current release:
- #1665
- #1695
- #1721 - added.

### CODE REVIEW

**Comment by SvenDowideit** [comment]
@RichardScothern  @dmp42 @dmcgowan @thaJeztah

File: docs/Dockerfile [hunk 1]
```diff
@@ -1,18 +1,9 @@
...
-RUN svn checkout https://github.com/docker/kitematic/trunk/docs /docs/content/kitematic
-RUN svn checkout https://github.com/docker/toolbox/trunk/docs /docs/content/toolbox
-RUN svn checkout https://github.com/docker/opensource/trunk/docs /docs/content/opensource
-
 ENV PROJECT=registry
 
 # To get the git info for this repo
 COPY . /src
-
+RUN rm -r /docs/content/$PROJECT/
```
Comment (thaJeztah) [inline docs/Dockerfile]: oh, I didn't cherry-pick this in the docker/docker repo - is it needed? (didn't think it was)

**Comment by thaJeztah** [comment]
LGTM (but would like a check from the maintainers here)

**Comment by andrewhsu** [comment]
If possible, I would like to have #1665 in the upcoming release of docker distribution so we can sync our internal usage of the registry with open source official releases.

**Comment by thaJeztah** [comment]
@andrewhsu this PR is only to update the documentation with fixes that were merged in master

**Comment by andrewhsu** [comment]
I see...how can I determine if/when the #1665 feature will go into an official release?

**Comment by thaJeztah** [comment]
If it's merged in "master", it will be in the next release; releases are branched of the master branch

File: docs/spec/api.md [hunk 2]
```diff
@@ -618,26 +618,6 @@ The "digest" parameter must be included with the PUT request. Please see the
 [_Completed Upload_](#completed-upload) section for details on the parameters
 and expected responses.
 
-Additionally, the upload can be completed with a single `POST` request to
```
Comment (stevvooe) [inline docs/spec/api.md]: Why is this being removed?

Comment (thaJeztah) [inline docs/spec/api.md]: See https://github.com/docker/distribution/pull/1702

File: docs/Dockerfile (hunk 1 above)
Comment (SvenDowideit) [inline docs/Dockerfile]: mmm, good point - it depends. in this case, I am, as it makes the process faster.

File: docs/spec/api.md (hunk 2 above)
Comment (SvenDowideit) [inline docs/spec/api.md]: @stevvooe I _am_ only guessing about what you guys want to update for the currently released docs-set.

would it be possible for you guys to adopt https://github.com/docker/docker/pull/22333 and thus label docs PR's that should go public?

Comment (stevvooe) [inline docs/spec/api.md]: I see.

@SvenDowideit Typically, distribution doesn't do cherry picking, as it wrecks history.

Comment (thaJeztah) [inline docs/spec/api.md]: You could limit those labels to just the "docs" part (to mark documentation changes / fixes that are safe to go out between releases, or apply to the current release)

Comment (SvenDowideit) [inline docs/spec/api.md]: @stevvooe cool - how do you guys update the docs for the release that's out there, and avoid releasing docs changes for vNext?

I'm happy to kick around ideas, what we're looking for in the end is something we can consistently apply to all repos with docs.

Comment (stevvooe) [inline docs/spec/api.md]: @SvenDowideit You'll have to ask @RichardScothern for the current release process.

I'd like to find something that allows us to release multiple versions of the docs and preserves the lineage between branches (something that cherry picks break).

Comment (SvenDowideit) [inline docs/spec/api.md]: at this point, we'd like to get all the repos onto the same process - and this one is the only outlier, meaning its also the only one that won't get automated docs updates.

once we use the same process, changing it is :+1: - but first I'd like to get docs updates out regularly.

**Comment by RichardScothern** [comment]
Commits are correct. #1721 is a correction and is also OK.

#1665, #1695 will be in the next release.

**Comment by SvenDowideit** [comment]
miss-reading @RichardScothern 's comment as a LGTM :)

**Pull request merged**